core/src/main/java/org/infinispan/interceptors/impl/DistCacheWriterInterceptor.java

```diff
@@ -87,6 +88,25 @@ public BasicInvocationStage visitPutKeyValueCommand(InvocationContext ctx, PutKe
    }
 
    @Override
+   public Object visitBackupWriteCommand(InvocationContext ctx, BackupWriteCommand command) throws Throwable {
+      return invokeNext(ctx, command).thenApply((rCtx, rCommand, rv) -> {
+         BackupWriteCommand backupWriteCommand = (BackupWriteCommand) rCommand;
+         if (!isStoreEnabled(backupWriteCommand))
+            return rv;
+
+         //no need to check for proper write. we are remote, in a backup owner.
```

Comment:
I think that would only work with `acquireSharedTopologyLock` (sorry, that's the one I meant, not the exclusive one). Without it you'd still have a chance of a topology change between the topology check and the actual write. If that happens, we could be writing an entry that we don't own any more after the stale entry invalidation. The command would be retried because EWI would notice the topology change during commit, but the entry would stay in the store.

While writing this I realized shared stores are even weirder, because we kind of assume exclusive access to each key from the primary owner, yet there can be multiple nodes thinking they're the primary owner at the same time. E.g. with a JDBC store that doesn't support upsert, I think we could search for the key, find 0 rows and decide to do an insert, and then fail it because the other primary owner already did the insert. There was a question on SO about primary key violation errors with a shared store and numOwners > 1, I wonder if their test involved something like this...